### PR TITLE
Implement pet effects as own #828

### PR DESCRIPTION
**Pull request by Zeffuro**
Implements option to set pet effects as your own effects. Implements #828 

https://puu.sh/ImXLw/122d9af816.mp4

### CODE REVIEW

File: DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs [hunk 1]
```diff
@@ -199,6 +206,28 @@ protected unsafe List<StatusEffectData> StatusEffectDataList(GameObject? actor)
...
 
+        protected bool IsStatusFromPlayerPet(StatusStruct status)
+        {
+            var player = Plugin.ClientState.LocalPlayer;
+            if (player == null)
+            {
+                return false;
+            }
+            
+            var character = Plugin.ObjectTable.SearchById(status.SourceID);
```
Comment (Tischel) [inline DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:217]: Doing this every frame for every status for every list is pretty costly.
I think you can directly find the pet's address/id by using the BuddyList class from Dalamud, but haven't tested it.

Comment (Zeffuro) [reply DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:217]: Well the only other options I thought off is including it in the StatusEffect struct but that required a whole more bunch of changes to if everywhere so wasn't sure what option would be best.

Comment (Tischel) [reply DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs:217]: I think the way you did it is fine if we can simplify the look up for the pet's actor.
I'm pretty sure you can get it directly with the class I mentioned above, but again, haven't really tested it.

Comment (Zeffuro) [reply DelvUI/Interface/StatusEffects/StatusEffectsListHud.cs]: Actually that was exactly what I needed, thanks for bringing that class up, works for me :)

**Review by Tischel — APPROVED** [review]

**Pull request merged**